### PR TITLE
✨ Start on user quickstart variant for testing

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR adds a variant user quickstart to use as an end-to-end test, sharing as much as possible with the other user quickstarts.

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
Just getting started. So far, just building and pre-positioning the core container image.

**Comment by MikeSpreitzer** [comment]
/cc @clubanderson 
/cc @francostellari 
/cc @dumb0002 
/cc @waltforme 
@yana1205

File: docs/content/common-subs/build-core-image.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<!--build-core-image-start-->
+Build the core image.
+
+```shell
+pwd
+export EXTRA_CORE_TAG=$(date +test%m%d-%H%M%S)
+make kubestellar-image-local
```
Comment (clubanderson) [inline docs/content/common-subs/build-core-image.md:7]: how long does this take on avg?  Takes forever for me for 3 images locally

File: docs/mkdocs.yml [hunk 2]
```diff
@@ -40,6 +40,7 @@ nav:
   - Getting Started:
       - Quickstart using Kind: Getting-Started/user-quickstart-kind.md 
       - Quickstart using OpenShift: Getting-Started/user-quickstart-openshift.md 
+      - Quickstart for testing: Getting-Started/user-quickstart-test.md
```
Comment (clubanderson) [inline docs/mkdocs.yml:43]: do we need to expose this?

File: docs/content/common-subs/build-core-image.md (hunk 1 above)
Comment (MikeSpreitzer) [reply docs/content/common-subs/build-core-image.md:7]: This only builds for one platform, the native one. Much faster than native + two emulated.

File: docs/mkdocs.yml (hunk 2 above)
Comment (MikeSpreitzer) [reply docs/mkdocs.yml:43]: I was wondering about that. Makes it easy for me to find and test. Maybe put in a contributor section?

**Comment by MikeSpreitzer** [comment]
@KPRoche verified that I do not know how to make `ko` push to `podman` (plain podman, as normally installed, no additional stuff) as if it were local Docker, when running on Linux. This is because podman on Linux does not _have_ a daemon, does not _have_ a Docker socket.

**Comment by MikeSpreitzer** [comment]
Successful test.

[qst-run1.log](https://github.com/kubestellar/kubestellar/files/13221661/qst-run1.log)

**Comment by MikeSpreitzer** [comment]
Oh, ugh. This adds a pre-req, `ko`.

File: docs/content/common-subs/build-core-image.md (hunk 1 above)
Comment (clubanderson) [reply docs/content/common-subs/build-core-image.md:7]: awesome - Thank you

File: docs/mkdocs.yml (hunk 2 above)
Comment (clubanderson) [reply docs/mkdocs.yml:43]: for now its fine - lets leave it, but we could move it later.  Not the most important detail.

**Review by clubanderson — COMMENTED** [review]
/lgtm

**Comment by clubanderson** [comment]
@MikeSpreitzer you'll have to close and re-open this.  The user-quickstart test I enabled this morning, and then shut off when we decided to go in this PRs direction, is stuck.

**Review by clubanderson — COMMENTED** [review]
/lgtm

great work here @MikeSpreitzer

**Comment by MikeSpreitzer** [comment]
The force-push was to prod testing after expanding allowed github actions.

**Comment by MikeSpreitzer** [comment]
Apparently the version of `make` that gets used (which is that? Why is `make` even installed?) here adds extra output, ruining the design of the xxx target.  Here is an excerpt from the failing workflow run:

```
make[1]: Leaving directory '/home/runner/work/kubestellar/kubestellar'
+++ make build-kubestellar-syncer-image-local
2023/10/31 23:44:17 Using base cgr.dev/chainguard/static:latest@sha256:d3465871ccaba3d4aefe51d6bb2222195850f6734cbbb6ef0dd7a3da49826159 for github.com/kubestellar/kubestellar/cmd/syncer
2023/10/31 23:44:17 Building github.com/kubestellar/kubestellar/cmd/syncer for linux/amd64
2023/10/31 23:45:13 Loading ghcr.io/kubestellar/kubestellar/syncer-3071a9d16bb776b2b5d229331bedb69d:0a2396954ea1ee17de0a5615e94959de2ee29014fcbe7e502ab442a0a8fdf66f
2023/10/31 23:45:14 Loaded ghcr.io/kubestellar/kubestellar/syncer-3071a9d16bb776b2b5d229331bedb69d:0a2396954ea1ee17de0a5615e94959de2ee29014fcbe7e502ab442a0a8fdf66f
2023/10/31 23:45:14 Adding tag latest
2023/10/31 23:45:14 Added tag latest
++ export 'SYNCER_IMG_REF=make[1]: Entering directory '\''/home/runner/work/kubestellar/kubestellar'\''
ghcr.io/kubestellar/kubestellar/syncer-3071a9d16bb776b2b5d229331bedb69d:0a2396954ea1ee17de0a5615e94959de2ee29014fcbe7e502ab442a0a8fdf66f
make[1]: Leaving directory '\''/home/runner/work/kubestellar/kubestellar'\'''
Locally built syncer image reference is <make[1]: Entering directory '/home/runner/work/kubestellar/kubestellar'
ghcr.io/kubestellar/kubestellar/syncer-3071a9d16bb776b2b5d229331bedb69d:0a2396954ea1ee17de0a5615e94959de2ee29014fcbe7e502ab442a0a8fdf66f
make[1]: Leaving directory '/home/runner/work/kubestellar/kubestellar'>; adding alias ko.local/syncer:test
++ SYNCER_IMG_REF='make[1]: Entering directory '\''/home/runner/work/kubestellar/kubestellar'\''
ghcr.io/kubestellar/kubestellar/syncer-3071a9d16bb776b2b5d229331bedb69d:0a2396954ea1ee17de0a5615e94959de2ee29014fcbe7e502ab442a0a8fdf66f
make[1]: Leaving directory '\''/home/runner/work/kubestellar/kubestellar'\'''
++ echo 'Locally built syncer image reference is <make[1]: Entering directory '\''/home/runner/work/kubestellar/kubestellar'\''
ghcr.io/kubestellar/kubestellar/syncer-3071a9d16bb776b2b5d229331bedb69d:0a2396954ea1ee17de0a5615e94959de2ee29014fcbe7e502ab442a0a8fdf66f
make[1]: Leaving directory '\''/home/runner/work/kubestellar/kubestellar'\''>; adding alias ko.local/syncer:test'
```

**Comment by MikeSpreitzer** [comment]
This has now got both images being build from local sources and those images being used. I think we can merge this and do more work in follow-on.

Remaining things to do:
- Use the local copy of the helm chart
- build and use a local "release" archive
- Use the local copy of the brew formula

**Pull request merged**